invoice.py

```diff
@@ -11,9 +11,27 @@
 class InvoiceLine(metaclass=PoolMeta):
     __name__ = 'account.invoice.line'
 
-    @fields.depends('party', 'invoice', 'quantity')
+    @fields.depends('product', 'party', 'invoice', 'quantity', 'invoice_type',
```

Comment:
Only add depends that require your method (on_change_product).
methods=[] is an empty list. You need add methods that use inside the on_change_product() method.

Example:

```python
@fields.depends(methods=['get_price_list'])
def on_change_product(self):
    Product = Pool().get('product.product')
    self.get_price_list()
    super(InvoiceLine, self).on_change_product()
```